TYPO3.Media/Classes/TYPO3/Media/Domain/Service/ImageService.php

```
@@ -143,8 +144,23 @@ public function processImage(FlowResource $originalResource, array $adjustments)
             $imagineImage = $this->applyAdjustments($imagineImage, $adjustments, $adjustmentsApplied);
         }
 
+        $qualityAdjustments = array_filter($adjustments, function ($v) {
+            return $v instanceof QualityImageAdjustment;
+        }, ARRAY_FILTER_USE_BOTH);
+        if (count($qualityAdjustments) > 0) {
+            /** @var QualityImageAdjustment $qualityAdjustment */
+            $qualityAdjustment = array_pop($qualityAdjustments);
+            $quality = $qualityAdjustment->getQuality();
+            if ($quality >= 1 && $quality <= 100) {
+                $additionalOptions['quality'] = $qualityAdjustment->getQuality();
+                $adjustmentsApplied = true;
+            }
```

hmm maybe throw a invalid configuration exception here? hard to debug otherwise